app/livechat/imports/server/rest/sms.js

```diff
@@ -98,33 +104,44 @@ API.v1.addRoute('livechat/sms-incoming/:service', {
 				visitorToken: token,
 			};
 
-			const uploadedFile = getUploadFile(details, smsUrl);
-			file = { _id: uploadedFile._id, name: uploadedFile.name, type: uploadedFile.type };
-
-			const fileUrl = FileUpload.getPath(`${ file._id }/${ encodeURI(file.name) }`);
-
-			const attachment = {
-				title: file.name,
-				type: 'file',
-				description: file.description,
-				title_link: fileUrl,
-			};
-
-			if (/^image\/.+/.test(file.type)) {
-				attachment.image_url = fileUrl;
-				attachment.image_type = file.type;
-				attachment.image_size = file.size;
-				attachment.image_dimensions = file.identify != null ? file.identify.size : undefined;
-			} else if (/^audio\/.+/.test(file.type)) {
-				attachment.audio_url = fileUrl;
-				attachment.audio_type = file.type;
-				attachment.audio_size = file.size;
-			} else if (/^video\/.+/.test(file.type)) {
-				attachment.video_url = fileUrl;
-				attachment.video_type = file.type;
-				attachment.video_size = file.size;
+			let attachment;
+			try {
+				const uploadedFile = getUploadFile(details, smsUrl);
+				file = { _id: uploadedFile._id, name: uploadedFile.name, type: uploadedFile.type };
+				const fileUrl = FileUpload.getPath(`${ file._id }/${ encodeURI(file.name) }`);
+
+				attachment = {
+					title: file.name,
+					type: 'file',
+					description: file.description,
+					title_link: fileUrl,
+				};
+
+				if (/^image\/.+/.test(file.type)) {
+					attachment.image_url = fileUrl;
+					attachment.image_type = file.type;
+					attachment.image_size = file.size;
+					attachment.image_dimensions = file.identify != null ? file.identify.size : undefined;
+				} else if (/^audio\/.+/.test(file.type)) {
+					attachment.audio_url = fileUrl;
+					attachment.audio_type = file.type;
+					attachment.audio_size = file.size;
+				} else if (/^video\/.+/.test(file.type)) {
+					attachment.video_url = fileUrl;
+					attachment.video_type = file.type;
+					attachment.video_size = file.size;
+				}
+			} catch (e) {
+				console.error(`Attachment upload failed: ${ e.message }`);
+				attachment = {
+					fields: [{
+						title: 'User upload failed',
```

Comment:
I think we should add translations for this. WDYT?

Reply:
I think it would be good 🤔 , what I think is that the FE won't translate message content, even if it looks like a translation string.

Reply:
For translations we usually use [this](https://github.com/RocketChat/Rocket.Chat/blob/3f2ce67aefb7b5bab2f44b5a9608729399d1fd30/app/utils/lib/tapi18n.js#L5) method which picks up the translations from the `i18n` folder. So earlier in my comment I was suggesting to add the translations within the translation files and modify this to something like
```suggestion
						title: t('sms_upload_failed'),
```

Reply:
🤔 but that would add a dependency on meteor for the code, and I know the team collab team is trying to reduce meteor's usage.

Reply:
ohh.. wasn't aware of this :confused: if that's the case, how are they proposing to handle the translations, any idea?

Reply:
Not actually as far as I know :(